### DISCUSSION TITLE
Use of axis surveillance camera for object detection

### DISCUSSION THREAD

**Discussion by udayzee05** [AxisCommunications/discussions#134] [General] [answered]
How to use AXIS M3057-PLVE Mk II Network Camera
 for object detection: I tried to access RTSP but got an assertion error and RTSP  failed to open. Can you please guide me through this? I am using YOLOv5 algorithm and python language.

**Comment by haichaoji123** [comment]
have you tried the url:  rtsp://root:pass@192.168.1.124/axis-media/media.amp?camera=1, for fisheye camera, there is different camera value to map to different view, it is better first try with VLC to see the URL work, then try to use other tools. if VLC can not work, try to factory default the camera. also try to use the latest firmware which you can download from Axis.com.

please refer to vapix for dewarped view:
https://www.axis.com/vapix-library/subjects/t10175981/section/t10011928/display?section=t10011928-t10011928

**Reply by udayzee05** [reply]
yes, I tried with VLC, and everything is working fine, but with FFmpeg and OpenCV it is throwing an Assertion error-Failed to open RTSP, I upgraded the firmware to the latest version, this is rtsp format I am using "rtsp://root:pass@192.168.0.1:554/axis-media/media.amp?camera=5"

**Reply by vivekatoffice** [reply]
Used **AXIS M3057-PLVE** Fixed Dome Network Camera with **AXIS OS version 10.12.130**:

1. Get list of the resolution supported using [VAPIX API](https://www.axis.com/vapix-library/subjects/t10175981/section/t10011928/display?section=t10011928-t10011937)

```sh
GET:  http://<ip>/axis-cgi/param.cgi?action=list&group=Properties.Image.I0

Response:

root.Properties.Image.I0.Mirroring=yes
root.Properties.Image.I0.Resolution=2048x2048,1920x1920,1728x1728,1440x1440,1280x1280,1200x1200,800x800,720x720,640x640,480x480,320x320,240x240,160x160
root.Properties.Image.I0.Rotation=0,180
root.Properties.Image.I0.ShowSuboptimalResolutions=false
```

2. **Test 2048x2048** resolution with YoloV5 [example](https://github.com/ultralytics/yolov5/blob/master/detect.py)

Command: python detectusingAxis.py --source "rtsp://root:<password>@<IP>/axis-media/media.amp?camera=1&resolution=2048x2048"

**Output**:
![image](https://user-images.githubusercontent.com/52032125/209065353-f47df5a9-2561-4ea5-8693-f5b7450c83fc.png)

**logs**:
```sh
detect: weights=yolov5s.pt, source=rtsp://root:<password>@<IP>/axis-media/media.amp?camera=1&resolution=2048x2048, data=data\coco128.yaml, imgsz=[640, 640], conf_thres=0.25, iou_thres=0.45, max_det=1000, device=, view_img=False, save_txt=False, save_conf=False, save_crop=False, nosave=False, classes=None, agnostic_nms=False, augment=False, visualize=False, update=False, project=runs\detect, name=exp, exist_ok=False, line_thickness=3, hide_labels=False, hide_conf=False, half=False, dnn=False, vid_stride=1
YOLOv5  v7.0-46-g96a71b1 Python-3.10.9 torch-1.13.1+cpu CPU

Fusing layers...
YOLOv5s summary: 213 layers, 7225885 parameters, 0 gradients
1/1: rtsp://root:<password>@<IP>/axis-media/media.amp?camera=1&resolution=2048x2048...  Success (inf frames 2048x2048 at 30.00 FPS)

0: 640x640 (no detections), 188.3ms
0: 640x640 (no detections), 188.8ms
0: 640x640 (no detections), 204.0ms...........................
```

3. **Test 720x720** resolution with YoloV5 [example](https://github.com/ultralytics/yolov5/blob/master/detect.py)

Command: python detectusingAxis.py --source "rtsp://root:<password>@<IP>/axis-media/media.amp?camera=1&resolution=720x720"

**Output**:
![image](https://user-images.githubusercontent.com/52032125/209066100-3c2bea79-5027-4f8d-81c1-d2b81613b381.png)

**logs**:
```sh
detect: weights=yolov5s.pt, source=rtsp://root:<password>@<IP>/axis-media/media.amp?camera=1&resolution=720x720, data=data\coco128.yaml, imgsz=[640, 640], conf_thres=0.25, iou_thres=0.45, max_det=1000, device=, view_img=False, save_txt=False, save_conf=False, save_crop=False, nosave=False, classes=None, agnostic_nms=False, augment=False, visualize=False, update=False, project=runs\detect, name=exp, exist_ok=False, line_thickness=3, hide_labels=False, hide_conf=False, half=False, dnn=False, vid_stride=1
YOLOv5  v7.0-46-g96a71b1 Python-3.10.9 torch-1.13.1+cpu CPU

Fusing layers...
YOLOv5s summary: 213 layers, 7225885 parameters, 0 gradients
1/1: rtsp://root:<password>@<IP>/axis-media/media.amp?camera=1&resolution=720x720...  Success (inf frames 720x720 at 30.00 FPS)

0: 640x640 (no detections), 141.8ms
0: 640x640 (no detections), 141.5ms
0: 640x640 (no detections), 125.1ms

```

@udayzee05  Please check if you have the same issue with the latest YoloV5 release too.

**Reply by vivekatoffice** [reply, marked as answer]
Used **AXIS M3057-PLVE Mk II Network Camera** with **AXIS OS version 11.1.72**:

1. Get list of the resolution supported using [VAPIX API](https://www.axis.com/vapix-library/subjects/t10175981/section/t10011928/display?section=t10011928-t10011937)

```sh
GET:  http://<ip>/axis-cgi/param.cgi?action=list&group=Properties.Image.I0

Response:

root.Properties.Image.I0.Mirroring=yes
root.Properties.Image.I0.Resolution=2016x2016,1920x1920,1728x1728,1440x1440,1280x1280,1200x1200,800x800,720x720,640x640,480x480,320x320,240x240,160x160
root.Properties.Image.I0.Rotation=0
root.Properties.Image.I0.ShowSuboptimalResolutions=false
```
2. **Test 2016x2016** resolution with YoloV5 [example](https://github.com/ultralytics/yolov5/blob/master/detect.py)

Command: python detectusingAxis.py --source "rtsp://root:<password>@<IP>/axis-media/media.amp?camera=1&resolution=2016x2016"

**Output**:
![image](https://user-images.githubusercontent.com/52032125/209075965-3788fa48-90db-4ca6-ac15-2165a12b2127.png)

2. **Test 640x640** resolution with YoloV5 [example](https://github.com/ultralytics/yolov5/blob/master/detect.py)

Command: python detectusingAxis.py --source "rtsp://root:<password>@<IP>/axis-media/media.amp?camera=1&resolution=640x640"

**Output**:
![image](https://user-images.githubusercontent.com/52032125/209079963-d8e4b2d4-1a62-4784-8383-4029102f3227.png)

**Outcome**:

Able to use AXIS M3057-PLVE Mk II Network Camera with Yolov5
